src/database/removeRemote.ts

```diff
@@ -1,41 +1,26 @@
-import { Response } from "request";
 import * as request from "request";
+import { Response } from "request";
 import * as responseToError from "../responseToError";
 import * as utils from "../utils";
 import * as FirebaseError from "../error";
 import * as logger from "../logger";
 import * as api from "../api";
 
-export enum NodeSize {
-  SMALL = "small",
-  LARGE = "large",
-  EMPTY = "empty",
-}
-
 export interface RemoveRemote {
   /**
-   *
-   * @param {string} path
    * @return {Promise<boolean>} true if the deletion is sucessful.
    */
   deletePath(path: string): Promise<boolean>;
 
   /**
-   *
-   * Run a prefetch test on a path before issuing a delete to detect
-   * large subtrees and issue recursive chunked deletes instead.
-   *
-   * @param {string} path
-   * @return {Promise<NodeSize>}j
+   * @return {Promise<boolean>} true if the deletion is sucessful.
    */
-  prefetchTest(path: string): Promise<NodeSize>;
+  deleteSubPath(path: string, children: string[]): Promise<boolean>;
 
   /**
-   *
-   * @param {string} path
    * @return {Promise<string[]>} the list of sub pathes found.
    */
-  listPath(path: string): Promise<string[]>;
+  listPath(path: string, numChildren: number): Promise<string[]>;
```

Comment:
Documentation is missing for `numChildren`

Reply:
Added